src/smt/proof_post_processor.cpp

```diff
@@ -157,12 +168,26 @@ bool ProofPostprocessCallback::updateInternal(Node res,
   return update(res, id, children, args, cdp, continueUpdate);
 }
 
+bool ProofPostprocessCallback::shouldExpand(PfRule id) const
+{
+  if (d_elimRules.find(id) != d_elimRules.end())
+  {
+    return true;
+  }
+  if (d_elimAllTrusted && d_pc->getPedanticLevel(id) > 0)
```

Comment:
do this test first?

Reply:
The latter is more expensive, technically, so this is the right order.